temporal-sdk/src/main/java/io/temporal/internal/statemachines/WorkflowStateMachines.java

```diff
@@ -267,6 +264,13 @@ private void handleSingleEvent(HistoryEvent event, boolean hasNextEvent) {
       return;
     }
 
+    if (replaying
+        && !hasNextEvent
+        && (event.getEventType() == EventType.EVENT_TYPE_WORKFLOW_TASK_STARTED
+            || WorkflowExecutionUtils.isWorkflowTaskClosedEvent(event))) {
+      replaying = false;
+    }
```

Comment:
Previously we were effectively switching it on WorkfowTaskScheduled, after it's processing (which was empty).
Now we switch it on WorkflowTaskStarted. It has to move up because the last WorkflowTaskStarted actually triggers an eventLoop and workflow code / state machines need to see a flipped flag.

Reply:
Got it